views/js/test/runner/sample/minimalisticProvider.js

```diff
@@ -0,0 +1,109 @@
+/*
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; under version 2
+ * of the License (non-upgradable).
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ *
+ * Copyright (c) 2015 (original work) Open Assessment Technologies SA ;
+ *
+ */
+/**
+ * @author Sam <sam@taotesting.com>
+ */
+define([], function(){
+
+    function initEvents(runner){
+
+        runner
+            .on('ready', function(){
+
+                //the plugins are ready
+
+                //the DOM and GUI are also ready
+
+                //render item to this.gui.itemContainer
+                this.renderContent();
```

Comment:
As the `renderContent` method is called each time an update is needed, maybe the name of this method is not the best choice. I think it can be confusing...